src/credentials/GroupDataProviderImpl.cpp

```diff
@@ -1606,12 +1607,11 @@ CHIP_ERROR GroupDataProviderImpl::SetKeySet(chip::FabricIndex fabric_index, cons
     for (size_t i = 0; i < in_keyset.num_keys_used; ++i)
     {
         ByteSpan epoch_key(in_keyset.epoch_keys[i].key, Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES);
-        uint8_t key[Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES];
-        MutableByteSpan key_span(key, sizeof(key));
+        MutableByteSpan key_span(keyset.operational_keys[i].value);
         ReturnErrorOnFailure(Crypto::DeriveGroupOperationalKey(epoch_key, key_span));
-        ReturnErrorOnFailure(Crypto::DeriveGroupSessionId(ByteSpan(key, sizeof(key)), keyset.operational_keys[i].hash));
+        ReturnErrorOnFailure(Crypto::DeriveGroupSessionId(key_span, keyset.operational_keys[i].hash));
+        VerifyOrReturnError(memcmp(kZeroKey, key_span.data(), sizeof(kZeroKey)), CHIP_ERROR_INTERNAL);
```

Comment:
OK, so now it's clear we are making sure `key_span` is not all zeros.  Then I have the following question:

1) Why are we making sure of that?  Where in the spec is an all-0 operational key disallowed?
2) Should we do this check before we use `key_span` to derive the session id?

Reply:
The argument here is that an all-zeroes operation key is a weak key, and must be the result of a faulty key derivation function. This is why the check is perform after the derivation.
I am not sure if a valid epoch key may result in an all-zeroes operational key, but even if such key exist, we may want to reject it, in the grounds that a weak operational key must never be used.

Reply:
The entire key space is valid.  If we disallow any keys, then we need error conditions on key inputs which creates a surface area for oracle attacks.